codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/handler/SockJsPaths.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.handler.codec.sockjs.handler;
+
+import io.netty.handler.codec.sockjs.transport.TransportType;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class consists exclusively of a static method that return a {@link SockJsPath} parsing
+ * a path.
+ * <p>
+ * A SockJS path consists of the following parts:
+ * http://server:port/prefix/serverId/sessionId/transport
+ */
+public final class SockJsPaths {
+
+    private static final Pattern SERVER_SESSION_PATTERN = Pattern.compile("^/([^/.]+)/([^/.]+)/([^/.]+)");
+    private static final SockJsPath INVALID_PATH = new InValidSockJsPath();
+
+    private SockJsPaths() {
+    }
+
+    /**
+     * Parses the specified path and extracts SockJS path parameters.
+     *
+     * @param path the path to be parsed
+     * @return {@link SockJsPath} the parsed SockJS path.
+     */
+    public static SockJsPath parse(final String path) {
+        final Matcher matcher = SERVER_SESSION_PATTERN.matcher(path);
+        return matcher.find() ?
+                new ValidSockJsPath(matcher.group(1), matcher.group(2), matcher.group(3)) : INVALID_PATH;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Avoid instantiation through private constructors from outside of the constructors class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AAccessorClassGeneration)